modules/javafx.controls/src/main/java/javafx/scene/control/skin/TableColumnHeader.java

```diff
@@ -645,9 +646,11 @@ protected void resizeColumnToFitContent(int maxRows) {
             Region r = (Region) n;
             padding = r.snappedLeftInset() + r.snappedRightInset();
         }
-
-        TableRow<T> tableRow = new TableRow<>();
-        tableRow.updateTableView(tv);
+        Callback<TableView<T>, TableRow<T>> rowFactory = tv.getRowFactory();
+        TableRow<T> tableRow = rowFactory != null ? rowFactory.call(tv) : new TableRow<>();
+        tableSkin.getChildren().add(tableRow);
+        tableRow.applyCss();
+        ((SkinBase<?>) tableRow.getSkin()).getChildren().add(cell);
```

Comment:
I don't think this is a safe cast. Instead, you probably should do an `instanceof SkinBase` check before

Reply:
You're right. A tablerow may have been created that returns a custom skin which is not necessarily derived from SkinBase. 
However, since we do not have a getChildren() method in such a custom skin we will be unable to proceed. We could now abandon altogether (another return) but that mean that resizeColumnToFitContent no longer works at all in the presence of custom rows. So I think the best solution in that case would be to ignore the rowFactory and use a default `new TableRow<>` again to preserve at least the old behaviour in such cases.
I'll try and improve the patch.